### PR TITLE
[CBRD-23960] Unallow multiple SQL statements when creating a single Statement object in server-side JDBC

### PR DESCRIPTION
**Pull request by hgryoo**
http://jira.cubrid.org/browse/CBRD-23960

This PR changed structures to manage only one statement
- `method_query_handler` has only one `query_result`
- Structures for communication are changed to have only one `query_result`
  - execute_info
  - make_outresult_info
- Server-side JDBC is also changed the same way
- When preparing a statement, the `method_query_handler` checks if the parsed SQL is multiple statements, and if so, send an error.

**Example**) please refer to the demo/TestChangedSpecTest.java file
```
test1:
Statement stmt = conn.createStatement();
stmt.execute("select * from t1; insert into t1 (b) values (1); select * from t1"); // throws exception

test2:
String query = "select * from t1; insert into t1 (b) values (1); select * from t1";
PreparedStatement stmt = conn.prepareStatement(query); // throws exception
```

NOTE
- please review from 51f1b7e

### CODE REVIEW

File: src/method/method_query_handler.cpp [hunk 1]
```diff
@@ -774,28 +601,18 @@ namespace cubmethod
...
     m_prepare_flag = flag;
 
-    m_current_result = NULL;
-    /* in cas_execute.c, srv_handle->cur_result_index = 0; is not actually index */
-    /* To make understandable, I'm not going to follow it */
-    m_current_result_index = -1;
-
-    query_result q_result;
+    query_result &q_result = m_query_result;
     q_result.stmt_type = stmt_type;
```
Comment (byungwook-kim) [inline src/method/method_query_handler.cpp:605]: stmt_type is always CUBRID_STMT_NONE. it looks useless.

File: src/method/method_query_handler.hpp [hunk 2]
```diff
@@ -111,13 +116,14 @@ namespace cubmethod
       int prepare_query (prepare_info &info, int &flag);
       int prepare_call (prepare_info &info, int &flag);
 
+      /* check */
+
+
```
Comment (ctshim) [inline src/method/method_query_handler.hpp]: There is a comment related to check, but no actual execution code.
Any code to add for check?

File: src/method/method_query_handler.cpp [hunk 3]
```diff
@@ -743,7 +570,7 @@ namespace cubmethod
 	db_session_set_xasl_cache_pinned (m_session, true, false);
       }
 
-    char stmt_type;
+    char stmt_type = CUBRID_STMT_NONE;
```
Comment (beyondykk9) [inline src/method/method_query_handler.cpp:580]: it looks not necessary. the stmt_type is always set by the folllowing code sequences.

File: src/method/method_query_handler.cpp [hunk 4]
```diff
@@ -246,218 +236,59 @@ namespace cubmethod
...
     int error = NO_ERROR;
-    int num_q_result = m_q_result.size();
     OID ins_oid = OID_INITIALIZER;
     DB_OBJECT *ins_obj_p;
 
-    qinfo.resize (num_q_result);
-    for (int i = 0; i < num_q_result; i++)
+    query_result &qresult = m_query_result;
+    DB_QUERY_RESULT *qres = (DB_QUERY_RESULT *) qresult.result;
+    if (qresult.stmt_type == CUBRID_STMT_INSERT && qres != NULL)
```
Comment (beyondykk9) [inline src/method/method_query_handler.cpp:249]: It needs an explanation why stmt_type must be CUBRID_STMT_INSERT.

File: src/method/method_query_handler.hpp (hunk 2 above)
Comment (hgryoo) [reply src/method/method_query_handler.hpp]: I'm gonna remove it.

File: src/method/method_query_handler.cpp (hunk 3 above)
Comment (hgryoo) [reply src/method/method_query_handler.cpp:580]: It is to suppress the uninitialized variable warnings.

Comment (beyondykk9) [reply src/method/method_query_handler.cpp:580]: Ok.

**Review by hyunikn — APPROVED** [review]

File: src/method/method_query_handler.cpp (hunk 4 above)
Comment (hgryoo) [reply src/method/method_query_handler.cpp:249]: It is for the GET_GENERATED_KEYS request. I've added a comment.

**Review by shparkcubrid — APPROVED** [review]

**Review by beyondykk9 — APPROVED** [review]

**Review by ctshim — APPROVED** [review]

**Review by youngjinj — APPROVED** [review]

**Pull request merged**